### PR TITLE
[WIP] v1alpha2 E2E tests

### PR DESCRIPTION
**Pull request by jadarsie**
I tried to use CAPI's E2E framework. It is nice but it did not quite work right now. The framework references `v1alpha3` types and the e2e module seems to be unusable at this point.

What I did for now is to temporarily copy the framework files while work is in progress.

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@jadarsie you might want to check out https://github.com/kubernetes-sigs/cluster-api/pull/1737

File: scripts/ci-e2e.sh [hunk 1]
```diff
@@ -17,7 +17,26 @@
...
+source "${REPO_ROOT}/hack/ensure-kustomize.sh"
+
+make test-e2e
+test_status="${?}"
+
+# TODO last chance to clean up resources if prow job leaves something behind
+
+exit "${test_status}"
+
+export CLIENT_ID=da8c7265-60fc-43bc-91b5-0aa0f1b05f25
```
Comment (alexeldeib) [inline scripts/ci-e2e.sh]: @jadarsie ?

Comment (alexeldeib) [reply scripts/ci-e2e.sh]: these look valid, i'd recommend rolling + removing from git history.

**Pull request closed without merging**